### PR TITLE
feat(transform): default to Manhattan transformations

### PR DESCRIPTION
**Pull request by rahulk29**
Transformations now refer to Manhattan transformations;
non-Manhattan transformations are currently unsupported.
Remove transformed views, since the Manhattan transformed view of
any type is itself.

### CODE REVIEW

File: libs/geometry/src/transform.rs [hunk 1]
```diff
@@ -456,7 +456,29 @@ fn unitary_matinv(a: &[[i8; 2]; 2]) -> [[i8; 2]; 2] {
...
+
+impl<T: TransformRef> TransformRef for Option<T> {
+    fn transform_ref(&self, trans: Transformation) -> Self {
+        self.as_ref().map(move |elt| elt.transform_ref(trans))
+    }
+}
+
+/// A trait for specifying how an object is changed by a [`Transformation`].
 #[impl_for_tuples(32)]
 pub trait TransformMut {
```
Comment (rohanku) [inline libs/geometry/src/transform.rs:483]: Why does TransformMut not need TranslateMut then?

File: libs/geometry/src/transform.rs [hunk 2]
```diff
@@ -456,7 +456,29 @@ fn unitary_matinv(a: &[[i8; 2]; 2]) -> [[i8; 2]; 2] {
...
+}
+
+impl<T: TransformRef> TransformRef for Vec<T> {
+    fn transform_ref(&self, trans: Transformation) -> Self {
+        self.iter().map(|elt| elt.transform_ref(trans)).collect()
+    }
+}
+
+impl<T: TransformRef> TransformRef for Option<T> {
+    fn transform_ref(&self, trans: Transformation) -> Self {
```
Comment (rohanku) [inline libs/geometry/src/transform.rs:476]: How do you decide if something should be `#[inline]` or not?

File: substrate/src/io/impls.rs [hunk 3]
```diff
@@ -653,19 +656,28 @@ where
     }
 }
 
+// TODO: Maybe do lazy translation here.
```
Comment (rohanku) [inline substrate/src/io/impls.rs:659]: How can you do lazy translation?

File: substrate/src/layout/mod.rs [hunk 4]
```diff
@@ -310,7 +307,7 @@ impl<T: ExportsLayoutData> Instance<T> {
     /// # Panics
     ///
     /// Panics if an error was thrown during generation.
-    pub fn cell(&self) -> Transformed<Cell<T>> {
+    pub fn cell(&self) -> TransformedCell<T> {
```
Comment (rohanku) [inline substrate/src/layout/mod.rs:310]: Why do we need TransformedCell?

**Review by rohanku — CHANGES_REQUESTED** [review]

File: libs/geometry/src/transform.rs (hunk 1 above)
Comment (rahulk29) [reply libs/geometry/src/transform.rs:483]: added TranslateMut as a supertrait of TransformMut (and similarly for Translate/Transform).

File: substrate/src/io/impls.rs (hunk 3 above)
Comment (rahulk29) [reply substrate/src/io/impls.rs:659]: removed comment

File: substrate/src/layout/mod.rs (hunk 4 above)
Comment (rahulk29) [reply substrate/src/layout/mod.rs:310]: removed

**Pull request merged**